node/src/proxy_server/mod.rs

```diff
@@ -2241,26 +2552,87 @@ mod tests {
-        assert_eq!(record, &expected_add_route_message);
+        let message = recording.get_record::<RouteResultMessage>(0);
+        assert_eq!(message, &expected_dns_retry_result_message);
+    }
+
+    #[test]
+    fn proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route() {
+        let test_name = "proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route";
+        let cryptde = main_cryptde();
+        let http_request = b"GET /index.html HTTP/1.1\r\nHost: nowhere.com\r\n\r\n";
+        let (proxy_server_mock, _, proxy_server_recording_arc) = make_recorder();
+        let route_query_response = None;
+        let (neighborhood_mock, _, _) = make_recorder();
+        let neighborhood_mock = neighborhood_mock
+            .stop_on_message(type_id!(RouteResultMessage))
+            .route_query_response(route_query_response.clone());
+        let socket_addr = SocketAddr::from_str("1.2.3.4:5678").unwrap();
+        let stream_key = make_meaningless_stream_key();
+        let expected_data = http_request.to_vec();
+        let msg_from_dispatcher = InboundClientData {
+            timestamp: SystemTime::now(),
+            peer_addr: socket_addr.clone(),
+            reception_port: Some(HTTP_PORT),
+            sequence_number: Some(0),
+            last_data: true,
+            is_clandestine: false,
+            data: expected_data.clone(),
+        };
+
+        let stream_key_factory = StreamKeyFactoryMock::new().make_result(stream_key);
+        let system = System::new(test_name);
+        let mut subject = ProxyServer::new(
+            cryptde,
+            alias_cryptde(),
+            true,
+            Some(STANDARD_CONSUMING_WALLET_BALANCE),
+            false,
+        );
+        subject.logger = Logger::new(test_name);
+        subject.stream_key_factory = Box::new(stream_key_factory);
+        let subject_addr: Addr<ProxyServer> = subject.start();
+        let mut peer_actors = peer_actors_builder()
+            .proxy_server(proxy_server_mock)
+            .neighborhood(neighborhood_mock)
+            .build();
+        // Get the dns_retry_result recipient so we can partially mock it...
+        let dns_retry_result_recipient = peer_actors.proxy_server.route_result_sub;
+        peer_actors.proxy_server = ProxyServer::make_subs_from(&subject_addr);
+        peer_actors.proxy_server.route_result_sub = dns_retry_result_recipient; //Partial mocking
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+
+        subject_addr.try_send(msg_from_dispatcher).unwrap();
+
+        system.run();
+
+        let recording = proxy_server_recording_arc.lock().unwrap();
+        let message = recording.get_record::<RouteResultMessage>(0);
+        assert_eq!(message.stream_key, stream_key);
+        assert_eq!(
+            message.result,
+            Err(format!(
+                "Failed to find route to nowhere.com for stream key: {stream_key}"
+            ))
+        );
```

Comment:
This is really puzzling me. 

You probably want to assert on the look of the message that arrived to Neighborhood, even though they might be identical. It would be a stronger test. 

However, here from just reading the tests, I don't understand why both actors would need to get this message. I was investigating on that and it seems like Neighborhood doesn't even have a Handler for this message!!! It's really weird. But it had to work out somehow because it does stop the test through the stop condition set at the top. I don't know. This is definitely weird. I was thinking, yeah, Neighborhood might not have the Handler of this message but we provide the Recorder which can handle every single message, still I don't know from where the message flew to it. 

Ohhhh. You know what?? I think it was the SystemKillerActor who allowed to finish the test. I think it's set to wait like 10 seconds probably and then it shut down the system. You can try out. If you have to be waiting really long it'll be an issue. You will want to get rid of the Neighborhood actor from this test and set the stop condition for the `proxy_server_mock`. That will ensure that you will get the message which you want to assert on afterwards. So you will end up really just with one assertion. However, check all these things out really carefully first.

Reply:
This is now fixed,
Adding a comment here to remind my-self to make sure this issue isn't happening with any other tests.